pkg/assembler/backends/ent/backend/artifact.go

```diff
@@ -138,3 +143,123 @@ func upsertArtifact(ctx context.Context, tx *ent.Tx, art *model.IDorArtifactInpu
 	}
 	return ptrfrom.String(artifactID.String()), nil
 }
+
+func (b *EntBackend) artifactNeighbors(ctx context.Context, nodeID string, allowedEdges edgeMap) ([]model.Node, error) {
+	var out []model.Node
+
+	query := b.client.Artifact.Query().
+		Where(artifactQueryPredicates(&model.ArtifactSpec{ID: &nodeID}))
+
+	if allowedEdges[model.EdgeArtifactHashEqual] {
+		query.
+			WithHashEqualArtA(func(q *ent.HashEqualQuery) {
+				getHashEqualObject(q)
+			}).
+			WithHashEqualArtB(func(q *ent.HashEqualQuery) {
+				getHashEqualObject(q)
+			})
+	}
+	if allowedEdges[model.EdgeArtifactIsOccurrence] {
+		query.
+			WithOccurrences(func(q *ent.OccurrenceQuery) {
+				getOccurrenceObject(q)
+			})
+	}
+	if allowedEdges[model.EdgeArtifactHasSbom] {
+		query.
+			WithSbom(func(q *ent.BillOfMaterialsQuery) {
+				getSBOMObject(q)
+			})
+	}
+	if allowedEdges[model.EdgeArtifactHasSlsa] {
+		query.
+			WithAttestations(func(q *ent.SLSAAttestationQuery) {
+				getSLSAObject(q)
+			}).
+			WithAttestationsSubject(func(q *ent.SLSAAttestationQuery) {
+				getSLSAObject(q)
+			})
+	}
+	if allowedEdges[model.EdgeArtifactCertifyVexStatement] {
+		query.
+			WithVex(func(q *ent.CertifyVexQuery) {
+				getVEXObject(q)
+			})
+	}
+	if allowedEdges[model.EdgeArtifactCertifyBad] {
+		query.
+			WithCertification(func(q *ent.CertificationQuery) {
+				getCertificationObject(q)
```

Comment:
worth filtering the certification's type with something like
```suggestion
				q.Where(certification.TypeEQ(certification.TypeBAD))
				getCertificationObject(q)
```

Reply:
yes that works. I will make the change

Reply:
done!